nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/smtp/SmtpConsumer.java

```diff
@@ -149,7 +149,7 @@ public void done() {
         }
 
         attributes.put("smtp.helo", context.getHelo());
-        attributes.put("smtp.remote.addr", context.getRemoteAddress().toString());
+        attributes.put("smtp.src", context.getRemoteAddress().toString().substring(1));
```

Comment:
Not sure why we need ```substring(1)``` here? Both ```toString()``` and ```substring(..)``` can cause exceptions (i.e., NPE and IndexArrayOutOfBound). 
I'll put a null check and remove ```substring```. Basically if at some point any formatting would be required for the address we have processors that  could do that down stream.